### PR TITLE
fix: Dependency updates

### PR DESCRIPTION
**Pull request by zhibek**
Update to latest packages (where possible, within Node version constraints, etc).

### CODE REVIEW

**Review by seb-cr — COMMENTED** [review]
All good apart from some ESLint stuff. Perhaps we could leave that for a separate PR. They're all dev dependencies, so if the goal is to remove vulnerabilities from services that use `lambda-wrapper` they would not affect this.

File: src/DependencyInjection/DependencyAware.class.js [hunk 1]
```diff
@@ -6,7 +6,6 @@ import DependencyInjection from './DependencyInjection.class';
 export default class DependencyAwareClass {
   /**
    * DependencyAwareClass constructor
-   *
```
[the remark this answers is not in the document]
Comment (Mahmoud-Ibrahim1) [reply src/DependencyInjection/DependencyAware.class.js]: just for record: the behavior of having blank line between JSDoc description and tags was forced by ```jsdoc/newline-after-description``` rule whic is no longer exist on the new version of ```eslint-plugin-jsdoc``` package was removed on 42.0.0 release, so in the new version we need to customize the rule  ```jsdoc/tag-lines``` to have a similar behavior like the old one .
 ```jsdoc/tag-lines: [ 'error', 'any', { startLines: 1 } ]```

File: package.json [hunk 2]
```diff
@@ -12,41 +12,41 @@
...
     "eslint-plugin-flowtype": "^8.0.3",
-    "eslint-plugin-import": "^2.25.2",
-    "eslint-plugin-jsdoc": "^39.3.2",
-    "eslint-plugin-sonarjs": "^0.13.0",
-    "eslint-plugin-unicorn": "^42.0.0",
-    "jest": "^28.1.3",
+    "eslint-plugin-import": "^2.28.0",
+    "eslint-plugin-jsdoc": "43.0.7",
+    "eslint-plugin-sonarjs": "^0.19.0",
+    "eslint-plugin-unicorn": "46.0.1",
```
[the remark this answers is not in the document]
Comment (Mahmoud-Ibrahim1) [reply package.json]: Done, reverted all esline.* packages updates and all the related eslint changes, as suggested we could leave that for a separate PR

File: src/DependencyInjection/DependencyAware.class.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (seb-cr) [reply src/DependencyInjection/DependencyAware.class.js]: Thanks for the explanation @Mahmoud-Ibrahim1 👍

**Review by seb-cr — APPROVED** [review]

**Pull request merged**